packages/compiler-cli/src/ngtsc/core/src/compiler.ts

```diff
@@ -1107,8 +1108,7 @@ export function isAngularCorePackage(program: ts.Program): boolean {
       return false;
     }
     // It must be exported.
-    if (stmt.modifiers === undefined ||
-        !stmt.modifiers.some(mod => mod.kind === ts.SyntaxKind.ExportKeyword)) {
+    if (!getModifiers(stmt)?.some(mod => mod.kind === ts.SyntaxKind.ExportKeyword)) {
```

Comment:
I think we could make this more readable without the optional chaining call:

```suggestion
    const modifiers = getModifiers(stmt);
    if (modifiers === undefined || !modifiers.some(mod => mod.kind === ts.SyntaxKind.ExportKeyword)) {
```
Otherwise I think this logic is very hard to follow, with the negation operating on both `undefined` and `false`. Not relying on negation of "falsy" but non-boolean values has been a code style point we've enforced across the compiler.

An alternative would be to make `getModifiers` return an empty array always, which would simplify downstream logic.